components/cam/src/physics/crm/sam/MICRO_SAM1MOM/cloud.F90

```diff
@@ -34,81 +35,83 @@ subroutine cloud(ncrms,q,qp,qn)
-            tabs1=(tabs(icrm,i,j,k)+fac1*qp(icrm,i,j,k))/(1.+fac2*qp(icrm,i,j,k))
-            ! Warm cloud:
-            if(tabs1.ge.tbgmax) then
-              tabs1=tabs(icrm,i,j,k)+fac_cond*qp(icrm,i,j,k)
-              qsatt = qsatw_crm(tabs1,pres(icrm,k))
-              ! Ice cloud:
-            elseif(tabs1.le.tbgmin) then
-              tabs1=tabs(icrm,i,j,k)+fac_sub*qp(icrm,i,j,k)
-              qsatt = qsati_crm(tabs1,pres(icrm,k))
-              ! Mixed-phase cloud:
-            else
-              om = an*tabs1-bn
-              qsatt = om*qsatw_crm(tabs1,pres(icrm,k))+(1.-om)*qsati_crm(tabs1,pres(icrm,k))
-            endif
-            !  Test if condensation is possible:
-            if(q(icrm,i,j,k).gt.qsatt) then
-              niter=0
-              dtabs = 100.
-              do while(abs(dtabs).gt.0.01.and.niter.lt.10)
-                if(tabs1.ge.tbgmax) then
-                  om=1.
-                  lstarn=fac_cond
-                  dlstarn=0.
-                  qsatt=qsatw_crm(tabs1,pres(icrm,k))
-                  dqsat=dtqsatw_crm(tabs1,pres(icrm,k))
-                else if(tabs1.le.tbgmin) then
-                  om=0.
-                  lstarn=fac_sub
-                  dlstarn=0.
-                  qsatt=qsati_crm(tabs1,pres(icrm,k))
-                  dqsat=dtqsati_crm(tabs1,pres(icrm,k))
-                else
-                  om=an*tabs1-bn
-                  lstarn=fac_cond+(1.-om)*fac_fus
-                  dlstarn=an*fac_fus
-                  qsatt=om*qsatw_crm(tabs1,pres(icrm,k))+(1.-om)*qsati_crm(tabs1,pres(icrm,k))
-                  dqsat=om*dtqsatw_crm(tabs1,pres(icrm,k))+(1.-om)*dtqsati_crm(tabs1,pres(icrm,k))
-                endif
-                if(tabs1.ge.tprmax) then
-                  omp=1.
-                  lstarp=fac_cond
-                  dlstarp=0.
-                else if(tabs1.le.tprmin) then
-                  omp=0.
-                  lstarp=fac_sub
-                  dlstarp=0.
-                else
-                  omp=ap*tabs1-bp
-                  lstarp=fac_cond+(1.-omp)*fac_fus
-                  dlstarp=ap*fac_fus
-                endif
-                fff = tabs(icrm,i,j,k)-tabs1+lstarn*(q(icrm,i,j,k)-qsatt)+lstarp*qp(icrm,i,j,k)
-                dfff=dlstarn*(q(icrm,i,j,k)-qsatt)+dlstarp*qp(icrm,i,j,k)-lstarn*dqsat-1.
-                dtabs=-fff/dfff
-                niter=niter+1
-                tabs1=tabs1+dtabs
-              end do
-              qsatt = qsatt + dqsat * dtabs
-              qn(icrm,i,j,k) = max(real(0.,crm_rknd),q(icrm,i,j,k)-qsatt)
-            else
-              qn(icrm,i,j,k) = 0.
-            endif
-            tabs(icrm,i,j,k) = tabs1
-            qp(icrm,i,j,k) = max(real(0.,crm_rknd),qp(icrm,i,j,k)) ! just in case
-          end do
+#elif defined(_OPENMP)
+    !$omp target teams distribute parallel do private(icrm,i,j,k,qsatt,dqsat,dtabs,tabs1,fff,dfff) 
+#endif
+    do iloop = 0, loop_size-1
```

Comment:
This is a pretty sweeping change to the kernels. If there's a bug we're working around in OpenMP, then please create one version of the new manually collapsed kernel for OpenMP4.5, and leave the original kernel there with OpenACC.